### DISCUSSION TITLE
Questions regarding how semantic tokens inherit their properties from more generic token classifications

### DISCUSSION THREAD

**Discussion by anthony-S93** [olimorris/onedarkpro.nvim#208] [Q&A] [answered]
The way in which semantic tokens inherit properties (via links to more generic tokens) seems a bit inconsistent to me. Here's a simple example: 
![2023-07-23-13-24-51](https://github.com/olimorris/onedarkpro.nvim/assets/69449791/f9481576-0fd3-4da4-a797-3ac08778fc9b)

Both `@lsp.mod.declaration.javascript` and `@lsp.typemod.class.declaration.javascript` are linked to more generic tokens `@lsp.mod.declaration` and `@lsp.typemod.class.declaration`, and rightly so since the former are language-specific whereas the latter are generic. This behavior makes sense, at least intuitively. 

`@lsp.type.class.javascript`, on the other hand, strays from this behavior.  Instead of being linked to `@lsp.type.class`, `@lsp.type.class.javascript` seems to have its values hard-coded. 

@olimorris Is this by design?

Doesn't it make more sense to apply a set of defaults to the generic tokens (such as `@lsp.type.class`, `@lsp.type.namespace`, etc.) and then link the language-specifc tokens to the generic ones? That way, we only need to add custom highlighting to lsp tokens when needed.

**Comment by olimorris** [comment, marked as answer]
> @olimorris Is this by design?

Yes it is. As per the [contributing](https://github.com/olimorris/onedarkpro.nvim/blob/main/CONTRIBUTING.md) guide, I reconciled back to the VS Code theme. The JavaScript semantic token highlight groups will have been added to ensure the theme's matched. If you notice any discrepancy, will merge a PR.

**Reply by anthony-S93** [reply]
Does VSCode assign different colors for Javascript classes and classes for other languages?

**Reply by olimorris** [reply]
I can't remember specifically. The VS Code theme does apply some specific highlights in certain places that give it a unique look; especially with JavaScript and TypeScript
